docs/development.adoc

```diff
@@ -179,234 +179,169 @@ $ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--ena
+    [...]
+})
 ----
 +
 
-* To run the tests sequentially or on single ginkgo test node use enviornment variable `TEST_EXEC_NODES`:
+* For a better understanding of what a spec does, use proper description in `Context` and `it` block
 +
 
-NOTE: However the env variable `TEST_EXEC_NODES` can be used to increase or decrease any number of test node for parallel run depending on the resource allocated to the OpenShift cluster. Make sure that the `TEST_EXEC_NODES` value cannot be `0` or any `negative` value.
-+
-
-Run component command integration tests
-+
 ----
-$ TEST_EXEC_NODES=1 make test-cmp-e2e
+For example:
+Context("when running help for storage command", func() {
+	It("should display the help", func() {
+		[...]
+	})
+})
 ----
 +
 
-Run application command integration tests
-+
-----
-$ TEST_EXEC_NODES=1 make test-cmd-app
-----
+* Due to parallel test run support make sure that the should run in isolation, otherwise the test result will lead to race condition. To achieve this ginkgo provides some in build functions `BeforeEach`, `AfterEach` etc.
 +
 
-Run storage command integration tests
-+
-----
-$ TEST_EXEC_NODES=1 make test-cmd-storage
 ----
-+
+For example:
+var _ = Describe("odo generic", func() {
+    var project string
+	var context string
+	var oc helper.OcRunner
+    BeforeEach(func() {
+	    oc = helper.NewOcRunner("oc")
+	    SetDefaultEventuallyTimeout(10 * time.Minute)
+	    context = helper.CreateNewContext()
+    })
+    AfterEach(func() {
+	    os.RemoveAll(context)
+    })
+    Context("deploying a component with a specific image name", func() {
+        JustBeforeEach(func() {
+            os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+            project = helper.CreateRandProject()
+        })
 
-Run watch command integration tests
-+
-----
-$ TEST_EXEC_NODES=1 make test-watch-e2e
+        JustAfterEach(func() {
+            helper.DeleteProject(project)
+            os.Unsetenv("GLOBALODOCONFIG")
+        })
+        It("should deploy the component", func() {
+            helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+            helper.CmdShouldPass("odo", "create", "nodejs:latest", "testversioncmp", "--project", project, "--context", context+"/nodejs-ex")
+            helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
+            helper.CmdShouldPass("odo", "delete", "-f", "--context", context+"/nodejs-ex")
+        })
+    })
+})
 ----
 +
 
-Run config/preference command integration tests
-+
-----
-$ TEST_EXEC_NODES=1 make test-odo-config
-----
+* Don’t create new test file for issues(bug) and try to add some scenario for each bug fix if applicable
 +
 
-Run generic integration tests
-+
-----
-$ TEST_EXEC_NODES=1 make test-generic
-----
+* Don’t use unnecessary text validation in `Expect` of certain command output. Only validation of key text specific to that scenario would be enough.
```

Comment:
@kadel Does it now make sense ?

Reply:
Still, don't know what you are trying to say :-(

Reply:
Don’t use unnecessary text validation in `Expect` of certain command output. In `Expect` only validation of key text specific of that scenario would be enough.
```
For example:
While running multiple push on same component without changing any source file.

// No text validation is required
helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
// only validating the particular text key to this scenario
output := helper.CmdShouldPass("odo", "push", "--context", context+"/nodejs-ex")
Expect(output).To(ContainSubstring("No file changes detected, skipping build"))
```

@kadel Please check, this could be helpful. You can suggest me a better phrase also